test/integration/work_test.go

```diff
@@ -949,21 +937,18 @@ var _ = ginkgo.Describe("ManifestWork", func() {
 			// Remove the delete option
 			gomega.Eventually(func() error {
 				work, err = hubWorkClient.WorkV1().ManifestWorks(o.SpokeClusterName).Get(context.Background(), work.Name, metav1.GetOptions{})
-
 				if err != nil {
 					return err
 				}
 
 				work.Spec.DeleteOption = nil
-				work, err = hubWorkClient.WorkV1().ManifestWorks(o.SpokeClusterName).Update(context.Background(), work, metav1.UpdateOptions{})
-
+				_, err = hubWorkClient.WorkV1().ManifestWorks(o.SpokeClusterName).Update(context.Background(), work, metav1.UpdateOptions{})
```

Comment:
if the `update` failed, we should not assign the value of `work`, otherwise, the work will be `nil`, and the `get` will be failed in the next **retry** since the `work.Name` is empty.